pkg/virt-handler/vm.go

```diff
@@ -1223,7 +1223,11 @@ func (d *VirtualMachineController) updatePausedConditions(vmi *v1.VirtualMachine
 	// Update paused condition in case VMI was paused / unpaused
 	if domain != nil && domain.Status.Status == api.Paused {
 		if !condManager.HasCondition(vmi, v1.VirtualMachineInstancePaused) {
-			calculatePausedCondition(vmi, domain.Status.Reason)
+			reason := domain.Status.Reason
+			if d.isVMIPausedDuringMigration(vmi) {
+				reason = api.ReasonPausedMigration
+			}
```

Comment:
I think it makes sense for this code to be moved into `calculatePausedCondition()`, and remove the `reason` argument.

Reply:
I think the reason is important in this case. However, one can distinguish between a user-initiated pause or migration controller.